src/pykeen/utils.py

```diff
@@ -509,6 +510,24 @@ def broadcast_cat(
     return torch.cat([x.repeat(*x_rep), y.repeat(*y_rep)], dim=dim)
 
 
+def multi_broadcast_cat(tensors: Iterable[torch.FloatTensor], dim: int) -> torch.FloatTensor:
+    """Apply :func:`broadcast_cat` to an arbitrary number of tensors.
+
+    :param tensors: A list of tensors to concatenate from left to right, using the same
+        semantics as :func:`torch.cat`
+    :param dim:
+        The concat dimension.
+
+    :return: A concatenated, broadcasted tensor.
+    """
+    tensors = iter(tensors)
+    first, second = next(tensors), next(tensors)
+    rv = broadcast_cat(first, second, dim=dim)
+    for tensor in tensors:
+        rv = broadcast_cat(rv, tensor, dim=dim)
+    return rv
```

Comment:
Not sure whether this version is better, but it might be easier to read.
```python
if len(tensors) == 0:
  raise ValueError
elif len(tensors) == 1:
  return tensors[0]
elif len(tensors) == 2:
  return broadcast_cat(tensors[0], tensors[1], dim=dim)
else:
  return broadcast_cat(tensors[0], multi_broadcast_cat(tensors[1:], dim=dim), dim=dim)

```

Reply:
ooooh i like this too. you wouldn't need a case for len(tensors)==2 though, since the recursive call would get a len==1 scenario

Reply:
nvm, the former one is (at least) as good.